rclpy/rclpy/executors.py

```diff
@@ -321,14 +321,24 @@ def spin_until_future_complete(
         future.add_done_callback(lambda x: self.wake())
 
         if timeout_sec is None or timeout_sec < 0:
-            while self._context.ok() and not future.done() and not self._is_shutdown:
+            while (
+                self._context.ok() and
+                not future.done() and
+                not future.cancelled()
+                and not self._is_shutdown
```

Comment:
small nit, can you keep the `and` location consistent. I think the new style recommendation is after the line break for easier readability.